front/app/containers/EventsPage/EventsViewer/index.tsx

```diff
@@ -51,47 +61,72 @@ const StyledPagination = styled(Pagination)`
 interface Props {
   title: string;
   fallbackMessage: string;
-  events: number[];
+  eventsTime: 'past' | 'future';
   className?: string;
 }
 
 const EVENTS_PER_PAGE = 10;
 
 const EventsViewer = memo<Props>(
-  ({ title, events, className, fallbackMessage }) => {
+  ({ title, fallbackMessage, eventsTime, className }) => {
+    const [projectIds, setProjectIds] = useState<string[]>([]);
+
+    const events = useEvents(
+      projectIds,
+      eventsTime === 'future',
+      eventsTime === 'past'
+    );
+
     const [currentPage, setCurrentPage] = useState<number>(1);
-    const [visibleEvents, setVisibleEvents] = useState<number[]>([]);
+    const [visibleEvents, setVisibleEvents] = useState<IEventData[]>([]);
```

Comment:
This should be handled by the API.

Reply:
Not the case currently... would require changes in the backend to fix

Reply:
See above comment, the useModerations hook needs adjustments